evm/src/memory/registers.rs

```diff
@@ -0,0 +1,45 @@
+//! Memory unit.
```

Comment:
Everything looks good here but I think some comments here would be useful, since readers might click through to the register definitions if they don't know what they represent. E.g. noting that the `_FIRST_CHANGE` ones are 1 iff that was the most significant address component that changed, otherwise 0. And that COUNTER just starts at 0 and increments by 1.

Reply:
Good idea!